From 6aa35a72d22989484789cb2a5c829d67de5b0f07 Mon Sep 17 00:00:00 2001
From: "jerome.zhao" <jerome.zhao@mediatek.com>
Date: Mon, 20 Feb 2023 14:44:31 +0800
Subject: [PATCH] jni_generator: Fix header guard with '-' in OUT_DIR

When generating header guards for source files with '-' in OUT_DIR,
character '-' is not allowed by Clang for a token after #ifndef.

error: ISO C99 requires whitespace after the macro name [-Werror,-Wc99-extensions]
#define _WORKTMP_OUT_AOSP-MASTER_GEN_COMPONENTS_CRONET_ANDROID_CRONET_JNI_REGISTRATION_

Change-Id: I4b69e3ac169b75cde0ca3a1956489f0193b621d1
---
 base/android/jni_generator/jni_registration_generator.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/base/android/jni_generator/jni_registration_generator.py b/base/android/jni_generator/jni_registration_generator.py
index 7b0aaa58d32b0..77b415c8f2785 100755
--- a/base/android/jni_generator/jni_registration_generator.py
+++ b/base/android/jni_generator/jni_registration_generator.py
@@ -103,8 +103,10 @@ def _Generate(java_file_paths,
         signature_to_cases, namespace)
 
   if header_path:
-    combined_dict['HEADER_GUARD'] = \
-        os.path.splitext(header_path)[0].replace('/', '_').replace('.', '_').upper() + '_'
+    header_guard = os.path.splitext(header_path)[0].upper() + '_'
+    for c in ('/', '.', '-'):
+        header_guard = header_guard.replace(c, '_')
+    combined_dict['HEADER_GUARD'] = header_guard
     combined_dict['NAMESPACE'] = namespace
     header_content = CreateFromDict(
         combined_dict,
-- 
2.40.0.rc1.284.g88254d51c5-goog

